standardtracer/propagation.go

```diff
@@ -177,52 +181,48 @@ func (p *splitBinaryPropagator) JoinTrace(
 	var traceID, propagatedSpanID int64
 	var sampledByte byte
 
-	err = binary.Read(contextReader, binary.BigEndian, &traceID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &traceID); err != nil {
 		return nil, opentracing.ErrTraceCorrupted
 	}
-	err = binary.Read(contextReader, binary.BigEndian, &propagatedSpanID)
-	if err != nil {
+	if err := binary.Read(contextReader, binary.BigEndian, &propagatedSpanID); err != nil {
```

Comment:
Yep. Ben used to have the concept of a TraceContextSource that was abstracting encoding from standard tracer. So it's not so much a nested package we need, as just an intermediate provider of injectors/extractors that the standard tracer would delegate to, and then derived implementations can just replace the provider (and reporter).